api/include/opentelemetry/logs/noop.h

```diff
@@ -34,11 +34,39 @@ class NoopLogger final : public Logger
 public:
   const nostd::string_view GetName() noexcept override { return "noop logger"; }
 
-  nostd::unique_ptr<LogRecord> CreateLogRecord() noexcept override { return nullptr; }
+  nostd::unique_ptr<LogRecord> CreateLogRecord() noexcept override
+  {
+    /*
+     * Do not return memory shared between threads,
+     * a `new` + `delete` for each noop record can not be avoided,
+     * due to the semantic of unique_ptr.
+     */
+    return nostd::unique_ptr<LogRecord>(new NoopLogRecord());
```

Comment:
Not sure if I fully understand the issue, but can we instead return a unique_ptr to static NoopLogRecord instance, with a custom  deleter which does nothing. This will ensure that the static instance is not deleted when the unique_ptr goes out of scope.
This way, if the LoggerProvider is not configured in SDK, and instrumented library won't do any allocation with every log call.

Reply:
this was explored in https://github.com/open-telemetry/opentelemetry-cpp/pull/2662 and did not work, unique_ptr damages vtable of a static instance.

Reply:
ok thanks for the link. Yes agree multiple delete calls from different threads would be an issue.